component/app.jsonnet

```diff
@@ -1,14 +1,25 @@
+local argocd = import 'lib/argocd.libjsonnet';
 local kap = import 'lib/kapitan.libjsonnet';
 local inv = kap.inventory();
-local params = inv.parameters.namespaces;
-local argocd = import 'lib/argocd.libjsonnet';
+local instance = inv.parameters._instance;
+
+// Prevent creating a non-instantiated instance
+assert instance != 'namespaces' : 'component must be instantiated with a name';
 
-local app = argocd.App('namespaces', params.namespace);
+local app = argocd.App(instance, '') {
```

Comment:
Can this parameter be empty or should it be `default`?

Reply:
I'm not sure what happens if it's empty, I'd set it to `default` (that's what we do in component adhoc-configurations).